src/control/server/ctl_storage.go

```diff
@@ -158,68 +157,46 @@ func (c *StorageControlService) PrepareNvme(req PrepareNvmeRequest) error {
 	)
 }
 
-type PrepareScmRequest struct {
-	Reset bool
-}
-
 // GetScmState performs required initialisation and returns current state
 // of SCM module preparation.
 func (c *StorageControlService) GetScmState() (types.ScmState, error) {
 	state := types.ScmStateUnknown
 
-	ok, _ := c.scm.ext.checkSudo()
+	ok, _ := c.ext.checkSudo()
 	if !ok {
 		return state, errors.Errorf("%s must be run as root or sudo", os.Args[0])
 	}
 
-	if err := c.scm.Setup(); err != nil {
-		return state, errors.WithMessage(err, "SCM setup")
-	}
-
-	if !c.scm.initialized {
-		return state, errors.New(msgScmNotInited)
+	if _, err := c.scm.Scan(scm.ScanRequest{Rescan: true}); err != nil {
```

Comment:
I missed this in my last review. Two things:

1. GetState() will automatically perform a scan if necessary, so it's not required to call this first.
2. Is the Rescan here intentional? This means that you're always forcing a rescan to service this RPC, which may be fine, but I thought the desire was to have the results cached.

Reply:
forcing rescan here is intentional, we want to be picking up changes from prepare etc. was going to revisit caching later